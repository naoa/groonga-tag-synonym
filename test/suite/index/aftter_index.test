register commands/tag_synonym

table_create Tags TABLE_PAT_KEY ShortText --default_tokenizer TokenDelimit
column_create Tags synonym COLUMN_SCALAR Tags

load --table Tags
[
{"_key": "Ruby", "synonym": "Rroonga"}
]

table_create Memos TABLE_HASH_KEY ShortText
column_create Memos tags COLUMN_VECTOR Tags

tag_synonym_add --table Memos --column tags
column_create Tags index COLUMN_INDEX Memos tags

load --table Memos
[
{"_key": "Groonga", "tags": "Groonga"},
{"_key": "Rroonga", "tags": "Groonga Ruby"}
]

select Memos --filter 'tags @ "Rroonga"' --output_columns 'tags'
select Tags
