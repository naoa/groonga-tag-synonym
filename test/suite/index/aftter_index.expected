register commands/tag_synonym
[[0,0.0,0.0],true]
table_create Tags TABLE_PAT_KEY ShortText --default_tokenizer TokenDelimit
[[0,0.0,0.0],true]
column_create Tags synonym COLUMN_SCALAR Tags
[[0,0.0,0.0],true]
load --table Tags
[
{"_key": "Ruby", "synonym": "Rroonga"}
]
[[0,0.0,0.0],1]
table_create Memos TABLE_HASH_KEY ShortText
[[0,0.0,0.0],true]
column_create Memos tags COLUMN_VECTOR Tags
[[0,0.0,0.0],true]
tag_synonym_add --table Memos --column tags
[[0,0.0,0.0],1]
column_create Tags index COLUMN_INDEX Memos tags
[[0,0.0,0.0],true]
load --table Memos
[
{"_key": "Groonga", "tags": "Groonga"},
{"_key": "Rroonga", "tags": "Groonga Ruby"}
]
[[0,0.0,0.0],2]
select Memos --filter 'tags @ "Rroonga"' --output_columns 'tags'
[[0,0.0,0.0],[[[0],[["tags","Tags"]]]]]
select Tags
[
  [
    0,
    0.0,
    0.0
  ],
  [
    [
      [
        5
      ],
      [
        [
          "_id",
          "UInt32"
        ],
        [
          "_key",
          "ShortText"
        ],
        [
          "index",
          "UInt32"
        ],
        [
          "synonym",
          "Tags"
        ]
      ],
      [
        4,
        "Groo",
        1,
        ""
      ],
      [
        3,
        "Groonga",
        1,
        ""
      ],
      [
        2,
        "Rroonga",
        0,
        ""
      ],
      [
        1,
        "Ruby",
        1,
        "Rroonga"
      ],
      [
        5,
        "nga",
        1,
        ""
      ]
    ]
  ]
]
